File: band2/band9/src/main/java/com/example/band9/BandApiService.java
package com.example.band9;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface BandApiService {
    @GET("v2/band/posts")
    Call<PostResponse> fetchPosts(
            @Query("access_token") String accessToken,
            @Query("band_key") String bandKey
    );

    @POST("v2/band/post/comment/create")
    Call<CommentResponse> postComment(
            @Query("access_token") String accessToken,
            @Query("band_key") String bandKey,
            @Query("post_key") String postKey,
            @Query("body") String body
    );
}

File: band2/band8/src/main/java/com/example/band8/BandApiService.kt
package com.example.band8

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface BandApiService {
    @GET("v2/band/posts")
    fun fetchPosts(
        @Query("access_token") accessToken: String,
        @Query("band_key") bandKey: String
    ): Call<PostResponse>

    @POST("v2/band/post/comment/create")
    fun postComment(
        @Query("access_token") accessToken: String,
        @Query("band_key") bandKey: String,
        @Query("post_key") postKey: String,
        @Query("body") body: String
    ): Call<CommentResponse>
}
